Azure/azure-cli
# [ARM] az ts:  Add support for --tags

**Description**

Template Specs create and update now supports --tags. 


**Testing Guide**

                               **The test cases added cover all of the following**

(1) 
az ts create -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}" --tag cli-test=test 

     Expected Result(s):

          If the parent template spec does not yet exist, the tag(s) will apply to both the parent (template spec) and the child 
          (template spec version). 

           If the parent template spec exists, only the child (template spec version) will be assigned the tag. Note: A child ( 
           template spec version) can have different tags from it's parent (template spec). Changing the tags of the child does not 
           affect the parent. 

(2) 

az ts create -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}"

       Expected Result(s): 
            
          If the parent (template spec) does not yet exist, no tags will be applied to the parent or the child (tags={}) 
          
          If the parent (template spec) exists then the new child will inherit the parents' tags or lack of tags. 

          If the child (template spec version) already exists this command will function the same. Create will override the child 
          (template spec version). 


(3a.)

az ts update -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}" --tags --yes 

az ts update -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}" --tags "" --yes

az ts update -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}" --tags  '' --yes 


     Expected Result(s):
          All the commands listed above in #3 should result in tags being removed (tags = {}). When updating a template 
          spec or template spec version tags should only be removed if explicitly empty. 

(3b.) 

az ts update -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}" --yes 

az ts update -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}"  --yes

az ts update -g {rg} -n {template_spec_name} -v 1.0 -l {resource_group_location} -f "{tf}" --yes 

     Expected Result(s):
          Tags should be unchanged. 


   

This checklist is used to make sure that common guidelines for a pull request are followed.
